### ISSUE TITLE
SA1101 respect static context

### ISSUE THREAD

**Issue by JohanLarsson** [DotNetAnalyzers/StyleCopAnalyzers#2656]
```cs
public class Foo
{
    public string Bar { get; } = nameof(Bar);
}
```

Should not warn here.
